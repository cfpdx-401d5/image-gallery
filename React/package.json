{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
