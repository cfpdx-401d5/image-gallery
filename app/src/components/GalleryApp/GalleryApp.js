import React, { Component } from 'react';
import ImageView from '../ViewSelector/ImageView';
import AddImage from '../AddImage/AddImage';
import ViewSelector from '../ViewSelector/ViewSelector';

import images from '../images/images';

const localUrl = 'http://localhost:8000/images';

export default class GalleryApp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            viewSelected: 'thumbnail',
            images: [],
        }
        this.handleClick = this.handleClick.bind(this);
        this.onAdd = this.onAdd.bind(this);
        this.onDelete = this.onDelete.bind(this);

    }// end constructor

    handleClick(value) {
        this.setState({ viewSelected: value })
    }

    onAdd(image) {
        console.log('image', image);
        fetch(localUrl, { headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(image), method: "POST" })
            .then(res => {
                return res.json();
            })
            .then((newImage) => {
                this.setState({
                    images: this.state.images.concat(newImage),
                });
            });
    }

    onDelete(toDelete) {
        fetch(`${localUrl}/${toDelete._id}`, { headers: { 'Content-Type': 'application/json' }, method: "DELETE" })
            .then(res => {
                return res.json();
            })
            .then((res) => {
                if (res.deleted) {
                    let newImages = this.state.images.filter((image) => {
                        if (image._id !== toDelete._id)
                            return image;
                    })
                    this.setState({
                        images: newImages
                    })
                }
            })
    }

    componentDidMount() {
        fetch(localUrl, {
            headers: { 'Content-Type': 'application/json' },
            method: "GET"
        })
            .then(res => {
                return res.json();
            })
            .then(images => {
                this.setState({ images })
            })
    }

    render() {
        return (
            <div >
                <ViewSelector handleClick={this.handleClick} />
                <AddImage onAdd={this.onAdd} />
                <ImageView viewSelected={this.state.viewSelected} images={this.state.images} onDelete={this.onDelete} />
            </div >
        );
    }
};