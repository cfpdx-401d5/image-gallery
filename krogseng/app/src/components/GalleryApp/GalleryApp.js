import React, { Component } from 'react';
import ImageView from '../ViewSelector/ImageView';
import AddImage from '../AddImage/AddImage';
import ViewSelector from '../ViewSelector/ViewSelector';
import images from '../images/images';

const localUrl = 'http://localhost:8000/images';

function UserGreeting(props) {
    return <h1>See what's new in the Gallery!</h1>;
}

function GuestGreeting(props) {
    return <h1>Please sign in to view the images.</h1>
}

function Greeting(props) {
    const isLoggedIn = props.isLoggedIn;
    if (isLoggedIn) {
        return <UserGreeting />;
    }
    return <GuestGreeting />;
}

function LoginButton(props) {
    return (
        <button onClick={props.onClick}>
            Login
            </button>
    );
}

function LogoutButton(props) {
    return (
        <button onClick={props.onClick}>
            Logout
            </button>
    );
}

class LoginControl extends React.Component {
    constructor(props) {
        super(props);
        this.handleLoginClick = this.handleLoginClick.bind(this);
        this.handleLogoutClick = this.handleLogoutClick.bind(this);
        this.state = { isLoggedIn: false };
    }

    handleLoginClick() {
        this.setState({ isLoggedIn: true });
    }

    handleLogoutClick() {
        this.setState({ isLoggedIn: false });
    }

    render() {
        const isLoggedIn = this.state.isLoggedIn;

        let button = null;
        if (isLoggedIn) {
            button = <LogoutButton onClick={this.handleLogoutClick} />;
        } else {
            button = <LoginButton onClick={this.handleLoginClick} />;
        }

        return (
            <div>
                <Greeting isLoggedIn={isLoggedIn} />
                {button}
            </div>
        );
    }
}

export default class GalleryApp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            viewSelected: 'thumbnail',
            images: [],
        }
        this.handleClick = this.handleClick.bind(this);
        this.onAdd = this.onAdd.bind(this);
        this.onDelete = this.onDelete.bind(this);

    }// end constructor

    handleClick(value) {
        this.setState({ viewSelected: value })
    }

    onAdd(image) {
        fetch(localUrl, { headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(image), method: "POST" })
            .then(res => {
                return res.json();
            })
            .then((newImage) => {
                this.setState({
                    images: this.state.images.concat(newImage),
                });
            });
    }

    onDelete(toDelete) {
        fetch(`${localUrl}/${toDelete._id}`, { headers: { 'Content-Type': 'application/json' }, method: "DELETE" })
            .then(res => {
                return res.json();
            })
            .then((res) => {
                if (res.deleted) {
                    let newImages = this.state.images.filter((image) => {
                        if (image._id !== toDelete._id)
                            return image;
                    })
                    this.setState({
                        images: newImages
                    })
                }
            })
    }

    componentDidMount() {
        fetch(localUrl, {
            headers: { 'Content-Type': 'application/json' },
            method: "GET"
        })
            .then(res => {
                return res.json();
            })
            .then(images => {
                this.setState({ images })
            })
    }

    render() {
        return (
            <div >
                <LoginControl />
                <ViewSelector handleClick={this.handleClick} />
                <AddImage onAdd={this.onAdd} />
                <ImageView viewSelected={this.state.viewSelected} images={this.state.images} onDelete={this.onDelete} />
            </div >
        );
    }
};