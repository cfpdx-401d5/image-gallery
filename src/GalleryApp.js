import React, { Component } from 'react';

let image = {
    title: 'Cute Bunny',
    description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
    url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
};

function ImageView(props) {
    if (props.viewSelected === 'listImage') {
        return (<ListImage image={image} />)
    } else if (props.viewSelected === 'thumbnail') {
        return (<Thumbnail image={image} />)
    } else if (props.viewSelected === 'gallery') {
        return (<Gallery image={image} />)
    }
}

function ListImage(props) {
    return (
        <div>
            <h2>Available Images</h2>
            <ul>
                <li>Title: {props.image.title}</li>
                <li><a href={props.image.url}>Link to Image </a></li>
                <li>Description: {props.image.description}</li>
            </ul>
        </div>
    );
}

function Thumbnail(props) {
    return (
        <div>
            <img alt={props.image.title} src={props.image.url} />
        </div>
    );
}

function Gallery(props) {
    return (
        <div>
            <img alt="this is the cutest bunny" src={props.image.url} />
        </div>
    );
}

function ViewSelect(props) {
    return (
        <div>
            <button onClick={() => props.onListImage()}>  List Images</button>
            <button onClick={() => props.onThumbnail()}> Thumbnail View</button>
            <button onClick={() => props.onGallery()}> Gallery View</button>
        </div>
    );
};

export default class GalleryApp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            viewChoice: 'listImage'
        }
        this.onListImage = this.onListImage.bind(this);
        this.onThumbnail = this.onThumbnail.bind(this);
        this.onGallery = this.onGallery.bind(this);

    }// end constructor
    onListImage() {
        this.setState({ viewChoice: 'listImage' })
    }
    onThumbnail() {
        this.setState({ viewChoice: 'thumbnail' })
    }
    onGallery() {
        this.setState({ viewChoice: 'gallery' })
    }

    render() {
        return (
            <div>
                <ViewSelect onListImage={this.onListImage}
                    onThumbnail={this.onThumbnail}
                    onGallery={this.onGallery}
                />
                <ImageView viewSelected={this.state.viewChoice} classname='viewer' />
            </div>
        );
    }
};
