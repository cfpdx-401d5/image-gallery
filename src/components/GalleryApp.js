import React, { Component } from 'react';
import views from './viewTypes';
import ImageView from './ImageView';

import images from './images';

function ViewSelect(props) {
    return (
        <div>
            {views.types.map(type => (
                <button
                    key={type.key}
                    onClick={() => props.handleClick(type.key)}
                >
                    {type.label}
                </button>
            ))}
        </div>
    );
};


function AddImage(props) {
    return (
        <form onSubmit={(e) => {
            e.preventDefault();
            const form = e.target;

            const els = form.elements;
            const image = {
                id: new Date().getTime(),
                title: els.title.value,
                description: els.description.value,
                url: els.url.value
            }

            props.onAdd(image);
            form.reset();
        }}>

            <label>Title
                <input name='title' />
            </label>
            <label>Description
                <input name='description' />
            </label>
            <label>URL
                <input name='url' />
            </label>
            <button type='submit'>Add </button>
        </form>
    );
}

export default class GalleryApp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            viewSelected: 'thumbnail',
            images: images,
        }
        this.handleClick = this.handleClick.bind(this);
        this.onAdd = this.onAdd.bind(this);

    }// end constructor

    handleClick(value) {
        this.setState({ viewSelected: value })
    }

    onAdd(image) {
        this.setState({
            images: this.state.images.concat(image),
        })
    }

    render() {
        return (
            <div>
                <AddImage onAdd={this.onAdd} />
                <ViewSelect handleClick={this.handleClick} />

                <ImageView viewSelected={this.state.viewSelected} classname='viewer' images={this.state.images} />
            </div>
        );
    }
};
