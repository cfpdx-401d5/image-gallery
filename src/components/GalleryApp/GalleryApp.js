import React, { Component } from 'react';
import ImageView from '../ViewSelector/ImageView';
import AddImage from '../AddImage/AddImage';
import ViewSelector from '../ViewSelector/ViewSelector';

import images from '../images/images';

export default class GalleryApp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            viewSelected: 'thumbnail',
            images: images,
        }
        this.handleClick = this.handleClick.bind(this);
        this.onAdd = this.onAdd.bind(this);
        this.onDelete = this.onDelete.bind(this);

    }// end constructor

    handleClick(value) {
        this.setState({ viewSelected: value })
    }

    onAdd(image) {
        this.setState({
            images: this.state.images.concat(image),
        })
    }

    onDelete(toDelete) {
        let newImages = this.state.images.filter((image) => {
            if (image.id !== toDelete.id)
                return image;
        })
        this.setState({
            images: newImages
        })
    }


    render() {
        return (
            <div>
                <ViewSelector handleClick={this.handleClick} />
                <AddImage onAdd={this.onAdd} />
                <ImageView viewSelected={this.state.viewSelected} images={this.state.images} onDelete={this.onDelete} />
            </div>
        );
    }
};
