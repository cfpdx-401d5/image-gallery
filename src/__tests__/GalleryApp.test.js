import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GalleryApp from '../components/GalleryApp';

describe('GalleryApp', () => {
    it('runs the GalleryApp', () => {
        const rendered = shallow(<GalleryApp />);
        const json = toJson(rendered);
        expect(json).toMatchSnapshot();
    })

});

