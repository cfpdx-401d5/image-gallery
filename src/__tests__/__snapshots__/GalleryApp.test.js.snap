exports[`GalleryApp runs the GalleryApp 1`] = `
<div>
  <ViewSelect
    handleClick={[Function]} />
  <ImageView
    classname="viewer"
    image={
      Object {
        "description": "Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?",
        "title": "Cute Bunny",
        "url": "http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg",
      }
    }
    viewSelected="listImage" />
</div>
`;
